x-pack/plugins/monitoring/server/telemetry_collection/get_logstash_stats.ts

```diff
@@ -116,20 +170,26 @@ export interface LogstashProcessOptions {
  */
 export function processStatsResults(
   results: estypes.SearchResponse<LogstashStats>,
-  { clusters, allEphemeralIds, versions, plugins }: LogstashProcessOptions
+  { clusters, allEphemeralIds, versions, plugins }: LogstashProcessOptions,
+  isSelfMonitoring: boolean = false
 ) {
   const currHits = results?.hits?.hits || [];
   currHits.forEach((hit) => {
-    const clusterUuid = hit._source!.cluster_uuid;
-    if (clusters[clusterUuid] === undefined) {
+    const clusterUuid = isSelfMonitoring
+      ? hit._source!.cluster_uuid
+      : hit._source!.logstash?.elasticsearch?.cluster?.id;
```

Comment:
review note: `cluster_uuid` is an alias for `logstash.elasticsearch.cluster.id` in the mapping but it only works when querying.